### DISCUSSION TITLE
Removing half colouring of bonds

### DISCUSSION THREAD

**Discussion by johncalpine** [rdkit/rdkit#6622] [Development]
When drawing structures with rdkit its colouring the half part of the connected bonds with the colour of the atom. Are there any options present in rdkit to turn off this half colouring without removing the atom label colour?

<img width="268" alt="Screenshot 2023-08-11 at 5 43 59 PM" src="https://github.com/rdkit/rdkit/assets/57757359/9351a444-0671-4dc9-adae-eef42f3d78d3">

<img width="261" alt="Screenshot 2023-08-11 at 5 45 42 PM" src="https://github.com/rdkit/rdkit/assets/57757359/7b6c69f6-7e64-4a68-88e5-718841f118f5">

**Comment by DavidACosgrove** [comment]
Hi,
There is no such option at the moment.  You could probably fake it with the
drawOptions highlightColour and highlightBondWidthMultiplier and passing
all the bonds in to DrawMolecule highlightBonds, with the colours in
highlightBondColors.  Not ideal, but it might get you up and running.  Note
the confusing mixture of "Colour" and "Color" in these options, for which I
can only apologise on behalf of my younger self.
It would be relatively straightforward to add a single colour bond option,
but there are already a large number of options and we are trying to avoid
adding to them.  What do you think, @Greg Landrum ***@***.***>?
Dave


On Fri, Aug 11, 2023 at 1:23 PM johncalpine ***@***.***>
wrote:

> When drawing structures with rdkit its colouring the half part of the
> connected bonds with the colour of the atom. Are there any options present
> in rdkit to turn off this half colouring without removing the atom label
> colour?
> [image: Screenshot 2023-08-11 at 5 43 59 PM]
> <https://user-images.githubusercontent.com/57757359/260004744-9351a444-0671-4dc9-adae-eef42f3d78d3.png> [image:
> Screenshot 2023-08-11 at 5 45 42 PM]
> <https://user-images.githubusercontent.com/57757359/260004796-7b6c69f6-7e64-4a68-88e5-718841f118f5.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/6622>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FSUFFJ4JXQU2LMZTUTXUYP2XANCNFSM6AAAAAA3M3AH6Q>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Comment by johncalpine** [comment]
Thanks Dave. Shall I expect this option in any of the future releases of rdkit?
